packages/base/src/tools.ts

```diff
@@ -212,3 +215,61 @@ export function createDefaultLayerRegistry(
     };
   }
 }
+
+// Get x/y tile values from lat and lng
+// Based on https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames#Mathematics
+function getTileCoordinates(latDeg: number, lonDeg: number, zoom: number) {
+  const latRad = latDeg * (Math.PI / 180);
+  const n = 1 << zoom;
+  const xTile = Math.floor(((lonDeg + 180.0) / 360.0) * n);
+  const yTile = Math.floor(
+    (n * (1 - Math.log(Math.tan(latRad) + 1 / Math.cos(latRad)) / Math.PI)) / 2
+  );
+
+  return { xTile, yTile };
+}
+
+export async function getLayerTileInfo(
+  tileUrl: string,
+  mapOptions: Pick<IJGISOptions, 'latitude' | 'longitude' | 'zoom'>,
+  urlParameters?: IDict<string>
+) {
+  // If it's tilejson, fetch the json to access the pbf url
+  if (tileUrl.includes('.json')) {
+    const response = await fetch(tileUrl);
+    if (!response.ok) {
+      throw new Error(`Failed to fetch json: ${response.statusText}`);
+    }
+
+    const json = await response.json();
+    tileUrl = json.tiles[0];
```

Comment:
Let's keep that for later us and open an issue for this, but there may be cases when there are multiple tiles urls, and we'll probably need to handle that case here.